### PR TITLE
Organize and annotate gitignores

### PR DESCRIPTION
**Pull request by rjernst**

### CODE REVIEW

**Comment by jpountz** [comment]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
+
+# testing stuff
+**/.local*
+.vagrant/
+
+# osx stuff
+.DS_Store
+
+# random old stuff that we should look at the necessity of...
+docs/html/
```
Comment (clintongormley) [inline .gitignore]: don't think we need this any more

File: .gitignore [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
+docs/build.log
+html_docs
```
Comment (clintongormley) [inline .gitignore]: i'd keep this as people frequently build the docs to check if they're ok

Comment (rjernst) [inline .gitignore]: Ok, kept that and added a comment to it.

**Pull request merged**